lib/topological_inventory/orchestrator/health_check.rb

```diff
@@ -0,0 +1,62 @@
+module TopologicalInventory
+  module Orchestrator
+    class HealthCheck
+      include Logging
+
+      attr_accessor :sources, :topology, :k8s
+
+      def initialize(sources, topology, k8s)
+        @sources = sources
+        @topology = topology
+        @k8s = k8s
+      end
+
+      def run
+        loop do
+          checks
+
+          sleep configured_seconds
+        end
+      end
+
+      def checks
+        errors = []
+
+        errors << check_url(topology)
+        errors << check_url(sources)
+        errors << check_k8s(k8s)
+
+        if errors.any?
+          FileUtils.rm_f("/tmp/healthy")
+        else
+          FileUtils.touch("/tmp/healthy")
+        end
+      end
+
+      private
+
+      def check_url(uri)
+        logger.info("[HealthCheck] Checking URL: #{uri}")
+
+        resp = Net::HTTP.get_response(uri.host, "/health", uri.port)
+        raise if resp.code.to_i != 200
```

Comment:
is it possible it'll be redirected (status 30x)? i.e. when new api version will be released

Reply:
I updated this to mimic the behavior of the default `http` livenessProbe in OpenShift. Basically now it'll register anything that is 400 (or higher) as error. That way if we get a redirect it's still fine.